src/main/java/org/threadly/concurrent/PriorityScheduler.java

```diff
@@ -1446,10 +1444,10 @@ public void cancel() {
     
     /**
      * Called as the task is being removed from the queue to prepare for execution.
+     * 
+     * @return true if the task should be executed
      */
-    public void executing() {
```

Comment:
This is what checks to ensure that the task CAN execute.  If this returns false we loop for a new task.  This is why all tasks take in the queue, to handle that they are able to atomically be removed (and thus ran)....as well as handling the rescheduling into the queue when they are done.